spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1159,10 +1158,23 @@ def stop_completion_services(self):
         logger.debug('Stopping completion services for %s' % self.filename)
         self.completions_available = False
 
-    @schedule_request(method=CompletionRequestTypes.DOCUMENT_DID_OPEN,
-                      requires_response=False)
+    @request(method=CompletionRequestTypes.DOCUMENT_DID_OPEN,
+             requires_response=False)
     def document_did_open(self):
         """Send textDocument/didOpen request to the server."""
+
+        # The connect is performed here instead of in __init__() because
+        # notify_close() may have been called (which disconnects the signal).
+        # Qt.UniqueConnection is used to avoid duplicate signal-slot connections
+        # (just in case).
+        #
+        # Note: PyQt5 throws if the signal is not unique (= already connected).
+        # It is an error if this happens because as per LSP specification
+        # `didOpen` “must not be sent more than once without a corresponding
+        # close notification send before”.
```

Comment:
This causes test failures. There are two ways two handle this
* Add `try … except` to ignore this “error”.
* Remove explicit `document_did_open()` calls in tests. It looks like they are not required (tests run locally), but I haven’t verified this.

Reply:
Please try to remove the `document_did_open()` calls from our tests. That would make things easier for us in the future.

Reply:
The force push at 2022-11-17T15:07+00:00 removes unnecessary calls of `document_did_open()`. There is still one test (`test_ipython_files`) which performs the call, but this is because the return value is tested.